core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -311,6 +312,31 @@ public void format() throws IOException {
         .toString());
   }
 
+  /**
+   * Triggers a checkpoint in the primary master ufs journal.
+   */
+  public synchronized void checkpoint() throws IOException {
+    long nextSequenceNumber = getNextSequenceNumberToWrite();
+    if (nextSequenceNumber - getNextSequenceNumberToCheckpoint() < ServerConfiguration.getLong(
+        PropertyKey.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES)) {
+      LOG.info("{}: No enough journal entries to checkpoint.",
```

Comment:
can we force a checkpoint anyway in this situation? The user is indicating that they want a checkpoint, so we should do our best to create one

Reply:
Now only when the `nextSequenceNumber == getNextSequenceNumberToCheckpoint()` we don't trigger snapshot.